internal/providers/testing/provider_mock.go

```diff
@@ -827,6 +838,70 @@ func (p *MockProvider) CallFunction(r providers.CallFunctionRequest) providers.C
 	return p.CallFunctionResponse
 }
 
+func (p *MockProvider) ValidateStateStoreConfig(r providers.ValidateStateStoreConfigRequest) (resp providers.ValidateStateStoreConfigResponse) {
+	p.Lock()
+	defer p.Unlock()
+
+	p.ValidateStateStoreConfigCalled = true
+	p.ValidateStateStoreConfigRequest = r
+
+	// Marshall the value to replicate behavior by the GRPC protocol,
+	// and return any relevant errors
+	storeSchema, ok := p.getProviderSchema().StateStores[r.TypeName]
+	if !ok {
+		resp.Diagnostics = resp.Diagnostics.Append(fmt.Errorf("no schema found for state store %q", r.TypeName))
+		return resp
+	}
+
+	_, err := msgpack.Marshal(r.Config, storeSchema.Body.ImpliedType())
+	if err != nil {
+		resp.Diagnostics = resp.Diagnostics.Append(err)
+		return resp
+	}
+
+	if p.ValidateStateStoreConfigFn != nil {
+		return p.ValidateStateStoreConfigFn(r)
+	}
+
+	if p.ValidateStateStoreConfigResponse != nil {
+		return *p.ValidateStateStoreConfigResponse
+	}
```

Comment:
This could be moved above the marshalling that replicates GRPC behaviour, to avoid any errors being returned in the test that expects a hardcoded response.

Similar feedback for `ConfigureStateStore` below